community-guidelines.md

```diff
@@ -0,0 +1,51 @@
+# Polyhack Community Guidelines
+
+**[#polyhack](https://github.com/polyhack)** is a community of
+programmers based on the IRC freenode network. It is specifically a
+polyglot group concerned to discuss programming ideas over the long-term.
+
+Its membership is predominantly from the Australian east coast but
+includes members from North America, New Zealand, and further. It was
+created in August, 2012.
+
+
+## Specific Guidelines
+
+ - Your guess is as good as mine
+
+
+
+## Bans
+
+There has been some discussion about the process for the rare case that
+someone exibits behaviour that is blatantly or consistently toxic to the
+community. The discussion appears primarily in
+[Issue 10](https://github.com/polyhack/helloworld/issues/10).
+
+There is not yet any formal process for handling bans. Some
+[discussion about a recent ban](https://github.com/polyhack/helloworld/issues/14)
+highlights the need for a more transparent process.
+
+
+### Proposed decision process for bans
+
+The following process is proposed in
+[a comment on Issue 14](https://github.com/polyhack/helloworld/issues/14#issuecomment-86754605).
+to deal with cases where someone's behaviour is not immediately or
+obviously abusive or disruptive.
+
+1. Create an issue here with reasonable details about the proposed ban.
+  - Refer to a specific community guideline or guidelines listed above
+    that are not being adhered to.
+  - Sometimes a guideline is understood implicitly by community members,
+    but is not written down. If the guideline being broken is not written
+    above, the issue discussion should include getting it added to this
+    document - that should happen before any ban goes ahead.
+  - Include chat logs.
+2. Keep the issue open until most regular members have had input and a
+   general consensus is reached, and the relevant guidelines have been
```

Comment:
I had in mind that no one is saying "WTF?" about it any more. If there is still back-and-forth discussion about something then it might not be right for the community. I would hope that for most things a consensus would be reached without the need for a vote, but there will always be something that is borderline. If discussion of an item drags on for weeks or months then it would probably be good to use a vote or some other formal mechanism to stop it dragging on forever.